test/stdlib/PrintFloat.swift.gyb

```diff
@@ -631,13 +631,23 @@ PrintTests.test("Printable_Double") {
 
   // Verify NaNs
   expectNaN("nan", Double.nan)
+  expectNaN("-nan", -Double.nan)
+#if arch(i386)
+  expectNaN("nan", Double(nan: 65535, signaling: false))
+  expectNaN("nan", Float64(bitPattern: 0x7fff_ffff_ffff_ffff))
+  expectNaN("nan", Float64(bitPattern: 0x7ffb_ffff_ffff_ffff))
+  expectNaN("-nan", -Double(nan: 65535, signaling: false))
```

Comment:
I expect this test is going to be fragile w.r.t. optimization on arm, where we run with the "Default NaN" bit set in FPCR, as that bit means that if the negation actually happens at runtime the resulting NaN won't be "negative".

More generally, IEEE 754 says that we shouldn't "interpret" the signbit of NaN (except for four operations: copy, copysign, negate, abs), so probably the best fix here is to simply print all NaNs as "nan" rather than "[-]nan". AFAIK we were only ever printing the leading `-` because it happened to fall out of `snprintf`.

Reply:
These are testing the _debug_ description, where sign, quiet bit, and payload have all been printed in the past—IMO that’s useful for debugging particularly because of platform-specific differences in NaN handling.

Reply:
I don't think it's actually very useful--its information about the encoding, not the value, which is  best accessed via `x.bitPattern`. My main concern though is that it's challenging to write portable tests for these cases.

I *do* think that we should be printing `snan` for signaling NaNs, though. And we should print the payload in the debug description.

Reply:
It’s of course possible to access everything about a NaN through its bit pattern, but FWIW I do use debug description to debug the sign of NaN. My point is that I find the result meaningful for debugging precisely _because_ it’s challenging to write portable tests.

I think if we’re going to debug print signaling vs. quiet and payload, then we’re wise to debug print the sign; after all, these are all aspects of NaN considered in the IEEE totalOrder operation.

Reply:
Printing `snan` is important because you're supposed to either signal invalid or print a different string--otherwise the signaling information is lost--and Swift doesn't model fenv, so we can't signal invalid. Printing the payload is less vital, but possibly useful for some people.

Printing the sign is of some interest to us as stdlib developers, but one else *should* care about it. It only has semantic meaning in `copysign` and the corresponding `init` and the totalOrder operation, which almost no one uses =)

I'm not going to say that we absolutely shouldn't print it, just that I don't think it carries its weight, and that it's leading us toward writing overly-restrictive tests.